### PR TITLE
Update4Camp (part 1/3)

### PR DESCRIPTION
**Pull request by Fergusson951**
▲From bottom to top▲

### CODE REVIEW

**Comment by Fergusson951** [comment]
if (marbleFollower()) helperArray[helperArray.length] = "Marble"; helpers++;
	if (followerHel()) helperArray[helperArray.length] = "Helia"; helpers++;
	if (followerKiha()) helperArray[helperArray.length] = "Kiha"; helpers++;
to
	if (marbleFollower()) {
		helperArray[helperArray.length] = "Marble";
		helpers++;
	}
	if (followerHel()) {
		helperArray[helperArray.length] = "Helia";
		helpers++;
	}
	if (followerKiha()) {
		helperArray[helperArray.length] = "Kiha";
		helpers++;
	}

**Comment by Fergusson951** [comment]
if (marbleFollower()) helperArray[helperArray.length] = "Marble" helpers++;
if (followerHel()) helperArray[helperArray.length] = "Helia" helpers++;
if (followerKiha()) helperArray[helperArray.length] = "Kiha" helpers++;
▓ this didn't work ▓

**Comment by Fergusson951** [comment]
In the end have moved descriptions

File: classes/classes/Scenes/Camp.as [hunk 1]
```diff
@@ -2604,47 +2578,31 @@ private function updateAchievements():void {
...
-		dungeonsCleared++;
-	}
-	if (kGAMECLASS.dungeons.checkSandCaveClear()) {
-		awardAchievement("Friend of the Sand Witches", kACHIEVEMENTS.DUNGEON_SAND_WITCH_FRIEND);
-		dungeonsCleared++;
-	}
+	if (kGAMECLASS.dungeons.checkFactoryClear()) awardAchievement("Shut Down Everything", kACHIEVEMENTS.DUNGEON_SHUT_DOWN_EVERYTHING); dungeonsCleared++;
+	if (kGAMECLASS.dungeons.checkDeepCaveClear()) awardAchievement("You're in Deep", kACHIEVEMENTS.DUNGEON_YOURE_IN_DEEP); dungeonsCleared++;
+	if (kGAMECLASS.dungeons.checkSandCaveClear()) awardAchievement("Friend of the Sand Witches", kACHIEVEMENTS.DUNGEON_SAND_WITCH_FRIEND); dungeonsCleared++;
+	if (kGAMECLASS.dungeons.checkLethiceStrongholdClear()) awardAchievement("End of Reign", kACHIEVEMENTS.DUNGEON_END_OF_REIGN); dungeonsCleared++;
```
Comment (Stadler76) [inline classes/classes/Scenes/Camp.as]: `if (expr) expr1; expr2;` is **not** the same as:
```ts
if (expr) {
	expr1;
	expr2;
}
```
!
Please fix/revert this.

File: classes/classes/Scenes/Camp.as [hunk 2]
```diff
@@ -2000,18 +1986,9 @@ private function buildCampWallPrompt():void {
...
-		helperArray[helperArray.length] = "Helia";
-		helpers++;
-	}
-	if (followerKiha()) {
-		helperArray[helperArray.length] = "Kiha";
-		helpers++;
-	}
+	if (marbleFollower()) helperArray[helperArray.length] = "Marble"; helpers++;
+	if (followerHel()) helperArray[helperArray.length] = "Helia"; helpers++;
+	if (followerKiha()) helperArray[helperArray.length] = "Kiha"; helpers++;
```
Comment (Stadler76) [inline classes/classes/Scenes/Camp.as]: `if (expr) expr1; expr2;` is **not** the same as:
```ts
if (expr) {
	expr1;
	expr2;
}
```
!
Please fix/revert this.

File: classes/classes/Scenes/Camp.as [hunk 3]
```diff
@@ -2604,47 +2578,31 @@ private function updateAchievements():void {
...
+	if (kGAMECLASS.dungeons.checkLethiceStrongholdClear()) awardAchievement("End of Reign", kACHIEVEMENTS.DUNGEON_END_OF_REIGN); dungeonsCleared++;
 	if (kGAMECLASS.dungeons.checkPhoenixTowerClear()) {
 		awardAchievement("Fall of the Phoenix", kACHIEVEMENTS.DUNGEON_PHOENIX_FALL);
 		dungeonsCleared++;
 		if (flags[kFLAGS.TIMES_ORGASMED] <= 0 && flags[kFLAGS.MOD_SAVE_VERSION] == kGAMECLASS.modSaveVersion) awardAchievement("Extremely Chaste Delver", kACHIEVEMENTS.DUNGEON_EXTREMELY_CHASTE_DELVER);
 	}
-	if (kGAMECLASS.dungeons.checkLethiceStrongholdClear()) {
-		awardAchievement("End of Reign", kACHIEVEMENTS.DUNGEON_END_OF_REIGN);
-		dungeonsCleared++;
-	}
```
Comment (Stadler76) [inline classes/classes/Scenes/Camp.as]: See above. Please fix/revert.

**Pull request merged**